{
  "uid" : "6b58929f9908fcbd",
  "name" : "test_topmenu_links",
  "fullName" : "tests.test_main_page.test_main_page.TestMainPage#test_topmenu_links",
  "historyId" : "eb18c938822b842c2adc92f9ac084549",
  "time" : {
    "start" : 1646393163040,
    "stop" : 1646393198272,
    "duration" : 35232
  },
  "description" : "\n        Проверка открытия страниц через главное меню\n        https://tfs.university.innopolis.ru/tfs/DefaultCollection/%D0%92%D0%BD%D1%83%D1%82%D1%80%D0%B5%D0%BD%D0%BD%D1%8F%D1%8F%20%D1%80%D0%B0%D0%B7%D1%80%D0%B0%D0%B1%D0%BE%D1%82%D0%BA%D0%B0/_workitems/edit/62193\n        ",
  "descriptionHtml" : "<pre><code>    Проверка открытия страниц через главное меню\n    https://tfs.university.innopolis.ru/tfs/DefaultCollection/%D0%92%D0%BD%D1%83%D1%82%D1%80%D0%B5%D0%BD%D0%BD%D1%8F%D1%8F%20%D1%80%D0%B0%D0%B7%D1%80%D0%B0%D0%B1%D0%BE%D1%82%D0%BA%D0%B0/_workitems/edit/62193\n</code></pre>\n",
  "status" : "failed",
  "statusMessage" : "selene.core.exceptions.TimeoutException: Message: \n\nTimed out after 10s, while waiting for:\nbrowser.element(('css selector', '#pagetitle')).has text ДМС\n\nReason: AssertionError: actual text: Абонемент в спорткомплекс «Иннополис»\nScreenshot: file:///home/seluser/.selene/screenshots/1646383817585/1646383817611.png\nPageSource: file:///home/seluser/.selene/screenshots/1646383817585/1646383817611.html",
  "statusTrace" : "self = <selene.core.wait.Wait object at 0x7f9f2e56a700>\nfn = <selene.core.conditions.ElementCondition object at 0x7f9f2e56a5e0>\n\n    def for_(self, fn: Callable[[E], R]) -> R:\n        finish_time = time.time() + self._timeout\n    \n        while True:\n            try:\n>               return fn(self._entity)\n\n/usr/local/lib/python3.8/dist-packages/selene/core/wait.py:102: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selene.core.conditions.ElementCondition object at 0x7f9f2e56a5e0>\nargs = (<selene.core.entity.Element object at 0x7f9f2e56a280>,), kwargs = {}\n\n    def __call__(self, *args, **kwargs):\n>       return self._fn(*args, **kwargs)\n\n/usr/local/lib/python3.8/dist-packages/selene/core/condition.py:136: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nentity = <selene.core.entity.Element object at 0x7f9f2e56a280>\n\n    def fn(entity: E) -> None:\n        query_to_str = str(query)\n        result = (\n            query.__name__\n            if query_to_str.startswith('<function')\n            else query_to_str\n        )\n        actual = query(entity)\n        if not predicate(actual):\n>           raise AssertionError(f'actual {result}: {actual}')\nE           AssertionError: actual text: Абонемент в спорткомплекс «Иннополис»\n\n/usr/local/lib/python3.8/dist-packages/selene/core/condition.py:109: AssertionError\n\nDuring handling of the above exception, another exception occurred:\n\nself = <tests.test_main_page.test_main_page.TestMainPage object at 0x7f9f2e4f7e50>\n\n    def test_topmenu_links(self):\n        \"\"\"\n        Проверка открытия страниц через главное меню\n        https://tfs.university.innopolis.ru/tfs/DefaultCollection/%D0%92%D0%BD%D1%83%D1%82%D1%80%D0%B5%D0%BD%D0%BD%D1%8F%D1%8F%20%D1%80%D0%B0%D0%B7%D1%80%D0%B0%D0%B1%D0%BE%D1%82%D0%BA%D0%B0/_workitems/edit/62193\n        \"\"\"\n        logger.info(\"Проверка открытия страниц через главное меню\")\n        admin_panel.auth_as_user(UsersConstants.FARID_GAYNULLIN)\n    \n        main_page.top_menu.click()\n        main_page.hr_help_menuitem.click()\n        main_page.page_title.have_text(\"HRhelp\")\n    \n        main_page.top_menu.click()\n        main_page.dms_menuitem.click()\n>       main_page.page_title.have_text(\"ДМС\")\n\ntests/test_main_page/test_main_page.py:166: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \npages/base_element.py:21: in have_text\n    self.locator.with_(Config(timeout=timeout)).should(have.text(text))\n/usr/local/lib/python3.8/dist-packages/selene/core/entity.py:379: in should\n    super().should(condition)\n/usr/local/lib/python3.8/dist-packages/selene/core/entity.py:122: in should\n    self.wait.for_(condition)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selene.core.wait.Wait object at 0x7f9f2e56a700>\nfn = <selene.core.conditions.ElementCondition object at 0x7f9f2e56a5e0>\n\n        def for_(self, fn: Callable[[E], R]) -> R:\n            finish_time = time.time() + self._timeout\n    \n            while True:\n                try:\n                    return fn(self._entity)\n                except Exception as reason:\n                    if time.time() > finish_time:\n    \n                        reason_message = str(reason)\n    \n                        reason_string = '{name}: {message}'.format(\n                            name=reason.__class__.__name__, message=reason_message\n                        )\n                        # todo: think on how can we improve logging failures in selene, e.g. reverse msg and stacktrace\n                        # stacktrace = getattr(reason, 'stacktrace', None)\n                        timeout = self._timeout\n                        entity = self._entity\n    \n                        failure = TimeoutException(\n                            f'''\n    \n    Timed out after {timeout}s, while waiting for:\n    {entity}.{fn}\n    \n    Reason: {reason_string}'''\n                        )\n    \n>                       raise self._hook_failure(failure)\nE                       selene.core.exceptions.TimeoutException: Message: \nE                       \nE                       Timed out after 10s, while waiting for:\nE                       browser.element(('css selector', '#pagetitle')).has text ДМС\nE                       \nE                       Reason: AssertionError: actual text: Абонемент в спорткомплекс «Иннополис»\nE                       Screenshot: file:///home/seluser/.selene/screenshots/1646383817585/1646383817611.png\nE                       PageSource: file:///home/seluser/.selene/screenshots/1646383817585/1646383817611.html\n\n/usr/local/lib/python3.8/dist-packages/selene/core/wait.py:125: TimeoutException",
  "flaky" : false,
  "newFailed" : false,
  "newBroken" : false,
  "newPassed" : false,
  "retriesCount" : 0,
  "retriesStatusChange" : false,
  "beforeStages" : [ {
    "name" : "_session_faker",
    "time" : {
      "start" : 1646383818661,
      "stop" : 1646383818786,
      "duration" : 125
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  }, {
    "name" : "auth_admin",
    "time" : {
      "start" : 1646383818786,
      "stop" : 1646383835082,
      "duration" : 16296
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  } ],
  "testStage" : {
    "description" : "\n        Проверка открытия страниц через главное меню\n        https://tfs.university.innopolis.ru/tfs/DefaultCollection/%D0%92%D0%BD%D1%83%D1%82%D1%80%D0%B5%D0%BD%D0%BD%D1%8F%D1%8F%20%D1%80%D0%B0%D0%B7%D1%80%D0%B0%D0%B1%D0%BE%D1%82%D0%BA%D0%B0/_workitems/edit/62193\n        ",
    "status" : "failed",
    "statusMessage" : "selene.core.exceptions.TimeoutException: Message: \n\nTimed out after 10s, while waiting for:\nbrowser.element(('css selector', '#pagetitle')).has text ДМС\n\nReason: AssertionError: actual text: Абонемент в спорткомплекс «Иннополис»\nScreenshot: file:///home/seluser/.selene/screenshots/1646383817585/1646383817611.png\nPageSource: file:///home/seluser/.selene/screenshots/1646383817585/1646383817611.html",
    "statusTrace" : "self = <selene.core.wait.Wait object at 0x7f9f2e56a700>\nfn = <selene.core.conditions.ElementCondition object at 0x7f9f2e56a5e0>\n\n    def for_(self, fn: Callable[[E], R]) -> R:\n        finish_time = time.time() + self._timeout\n    \n        while True:\n            try:\n>               return fn(self._entity)\n\n/usr/local/lib/python3.8/dist-packages/selene/core/wait.py:102: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selene.core.conditions.ElementCondition object at 0x7f9f2e56a5e0>\nargs = (<selene.core.entity.Element object at 0x7f9f2e56a280>,), kwargs = {}\n\n    def __call__(self, *args, **kwargs):\n>       return self._fn(*args, **kwargs)\n\n/usr/local/lib/python3.8/dist-packages/selene/core/condition.py:136: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nentity = <selene.core.entity.Element object at 0x7f9f2e56a280>\n\n    def fn(entity: E) -> None:\n        query_to_str = str(query)\n        result = (\n            query.__name__\n            if query_to_str.startswith('<function')\n            else query_to_str\n        )\n        actual = query(entity)\n        if not predicate(actual):\n>           raise AssertionError(f'actual {result}: {actual}')\nE           AssertionError: actual text: Абонемент в спорткомплекс «Иннополис»\n\n/usr/local/lib/python3.8/dist-packages/selene/core/condition.py:109: AssertionError\n\nDuring handling of the above exception, another exception occurred:\n\nself = <tests.test_main_page.test_main_page.TestMainPage object at 0x7f9f2e4f7e50>\n\n    def test_topmenu_links(self):\n        \"\"\"\n        Проверка открытия страниц через главное меню\n        https://tfs.university.innopolis.ru/tfs/DefaultCollection/%D0%92%D0%BD%D1%83%D1%82%D1%80%D0%B5%D0%BD%D0%BD%D1%8F%D1%8F%20%D1%80%D0%B0%D0%B7%D1%80%D0%B0%D0%B1%D0%BE%D1%82%D0%BA%D0%B0/_workitems/edit/62193\n        \"\"\"\n        logger.info(\"Проверка открытия страниц через главное меню\")\n        admin_panel.auth_as_user(UsersConstants.FARID_GAYNULLIN)\n    \n        main_page.top_menu.click()\n        main_page.hr_help_menuitem.click()\n        main_page.page_title.have_text(\"HRhelp\")\n    \n        main_page.top_menu.click()\n        main_page.dms_menuitem.click()\n>       main_page.page_title.have_text(\"ДМС\")\n\ntests/test_main_page/test_main_page.py:166: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \npages/base_element.py:21: in have_text\n    self.locator.with_(Config(timeout=timeout)).should(have.text(text))\n/usr/local/lib/python3.8/dist-packages/selene/core/entity.py:379: in should\n    super().should(condition)\n/usr/local/lib/python3.8/dist-packages/selene/core/entity.py:122: in should\n    self.wait.for_(condition)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selene.core.wait.Wait object at 0x7f9f2e56a700>\nfn = <selene.core.conditions.ElementCondition object at 0x7f9f2e56a5e0>\n\n        def for_(self, fn: Callable[[E], R]) -> R:\n            finish_time = time.time() + self._timeout\n    \n            while True:\n                try:\n                    return fn(self._entity)\n                except Exception as reason:\n                    if time.time() > finish_time:\n    \n                        reason_message = str(reason)\n    \n                        reason_string = '{name}: {message}'.format(\n                            name=reason.__class__.__name__, message=reason_message\n                        )\n                        # todo: think on how can we improve logging failures in selene, e.g. reverse msg and stacktrace\n                        # stacktrace = getattr(reason, 'stacktrace', None)\n                        timeout = self._timeout\n                        entity = self._entity\n    \n                        failure = TimeoutException(\n                            f'''\n    \n    Timed out after {timeout}s, while waiting for:\n    {entity}.{fn}\n    \n    Reason: {reason_string}'''\n                        )\n    \n>                       raise self._hook_failure(failure)\nE                       selene.core.exceptions.TimeoutException: Message: \nE                       \nE                       Timed out after 10s, while waiting for:\nE                       browser.element(('css selector', '#pagetitle')).has text ДМС\nE                       \nE                       Reason: AssertionError: actual text: Абонемент в спорткомплекс «Иннополис»\nE                       Screenshot: file:///home/seluser/.selene/screenshots/1646383817585/1646383817611.png\nE                       PageSource: file:///home/seluser/.selene/screenshots/1646383817585/1646383817611.html\n\n/usr/local/lib/python3.8/dist-packages/selene/core/wait.py:125: TimeoutException",
    "steps" : [ {
      "name" : "Авторизоваться на портале под ответственным пользователем",
      "time" : {
        "start" : 1646393163040,
        "stop" : 1646393174415,
        "duration" : 11375
      },
      "status" : "passed",
      "steps" : [ {
        "name" : "Авторизоваться под администратором портала",
        "time" : {
          "start" : 1646393163040,
          "stop" : 1646393163664,
          "duration" : 624
        },
        "status" : "passed",
        "steps" : [ ],
        "attachments" : [ ],
        "parameters" : [ ],
        "stepsCount" : 0,
        "attachmentsCount" : 0,
        "shouldDisplayMessage" : false,
        "hasContent" : false
      } ],
      "attachments" : [ ],
      "parameters" : [ {
        "name" : "user",
        "value" : "Фарид Гайнуллин"
      } ],
      "stepsCount" : 1,
      "attachmentsCount" : 0,
      "shouldDisplayMessage" : false,
      "hasContent" : true
    }, {
      "name" : "Screen-shot done",
      "time" : {
        "start" : 1646393198313,
        "stop" : 1646393198879,
        "duration" : 566
      },
      "status" : "passed",
      "steps" : [ ],
      "attachments" : [ {
        "uid" : "f878a4cfac213d2d",
        "name" : "screenshot",
        "source" : "f878a4cfac213d2d.png",
        "type" : "image/png",
        "size" : 1009924
      } ],
      "parameters" : [ ],
      "stepsCount" : 0,
      "attachmentsCount" : 1,
      "shouldDisplayMessage" : false,
      "hasContent" : true
    } ],
    "attachments" : [ {
      "uid" : "b45bb65000bc6103",
      "name" : "log",
      "source" : "b45bb65000bc6103.txt",
      "type" : "text/plain",
      "size" : 108
    } ],
    "parameters" : [ ],
    "stepsCount" : 3,
    "attachmentsCount" : 2,
    "shouldDisplayMessage" : true,
    "hasContent" : true
  },
  "afterStages" : [ {
    "name" : "auth_admin::0",
    "time" : {
      "start" : 1646401380670,
      "stop" : 1646401380911,
      "duration" : 241
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  } ],
  "labels" : [ {
    "name" : "tag",
    "value" : "main_page"
  }, {
    "name" : "parentSuite",
    "value" : "tests.test_main_page"
  }, {
    "name" : "suite",
    "value" : "test_main_page"
  }, {
    "name" : "subSuite",
    "value" : "TestMainPage"
  }, {
    "name" : "host",
    "value" : "portal-test-qa-nth69"
  }, {
    "name" : "thread",
    "value" : "6-MainThread"
  }, {
    "name" : "framework",
    "value" : "pytest"
  }, {
    "name" : "language",
    "value" : "cpython3"
  }, {
    "name" : "package",
    "value" : "tests.test_main_page.test_main_page"
  }, {
    "name" : "resultFormat",
    "value" : "allure2"
  } ],
  "parameters" : [ ],
  "links" : [ ],
  "hidden" : false,
  "retry" : false,
  "extra" : {
    "severity" : "normal",
    "retries" : [ ],
    "categories" : [ {
      "name" : "Product defects",
      "matchedStatuses" : [ ],
      "flaky" : false
    } ],
    "tags" : [ "main_page" ]
  },
  "source" : "6b58929f9908fcbd.json",
  "parameterValues" : [ ]
}